src/lib.rs

```diff
@@ -573,8 +573,7 @@ pub struct Type {
 }
 
 /// Enum with additional information, depending on the kind of type.
-#[derive(Debug, Eq, Hash, PartialEq)]
-#[cfg_attr(feature = "clone", derive(Clone))]
+#[derive(Debug, Clone, Eq, Hash, PartialEq)]
```

Comment:
I think we can get by without having `TypeInner` implement `Clone`.

It looks like this was only needed because of `ConcreteConstructor`.

Reply:
`ExpressionContext::resolve_type` also does require cloning, and I'm not sure how I can get around that.

Reply:
`TypeResolution` has a `Clone` impl. See how it's used here: https://github.com/gfx-rs/naga/blob/1650581fe7a94ed40067313f3cd205414dd4febf/src/front/glsl/types.rs#L296

Consider adding a `Typifier.get_handle` fn; could be used by both frontends.